src/node-file-trace.ts

```diff
@@ -363,15 +364,27 @@ export class Job {
           return;
         }
         if (Array.isArray(resolved)) {
-          for (const item of resolved) {
+          for (let item of resolved) {
             // ignore builtins
             if (item.startsWith('node:')) return;
+            // Dependencies which differ only by querystring are considered separate dependencies by
+            // webpack, and therefore need to be treated separately here, too. Resolution can strip
+            // the querystring off, in which case we need to put it back on to preserve the distinction.
+            if (queryString && !item.includes(queryString)) {
```

Comment:
Won't we accidentally skip adding the query string back if the query string itself is included in the path? e.g: `./fruit/apples?fruit`